scipy/sparse/linalg/eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -216,14 +217,54 @@ def test_hermitian():
         else:
             B = np.random.rand(size, size) + 1.j * np.random.rand(size, size)
             B = 10 * np.eye(size) + B.dot(B.T.conj())
-            w, v = lobpcg(H, X, B, maxiter=5000)
+            w, v = lobpcg(H, X, B, maxiter=5000, largest=False)
             w0, v0 = eigh(H, B)
 
         for wx, vx in zip(w, v.T):
             # Check eigenvector
-            assert_allclose(np.linalg.norm(H.dot(vx) - B.dot(vx) * wx) / np.linalg.norm(H.dot(vx)),
+            assert_allclose(np.linalg.norm(H.dot(vx) - B.dot(vx) * wx)
+                            / np.linalg.norm(H.dot(vx)),
                             0, atol=5e-4, rtol=0)
 
             # Compare eigenvalues
             j = np.argmin(abs(w0 - wx))
             assert_allclose(wx, w0[j], rtol=1e-4)
+
+def test_eigs_consistency():
+    n = 20
+    vals = np.arange(1, n+1, dtype=np.float64)
+    A = spdiags(vals, 0, n, n)
+    X = np.random.rand(n, 2)
+    lvals20, lvecs20 = lobpcg(A, X, largest=True, maxiter=100)
+    vals20, vecs20 = eigs(A, k=2)
+
+    _check_eigen(A, lvals20, lvecs20, atol=1e-3, rtol=0)
+    assert_allclose(vals20, lvals20, atol=1e-14)
+
+    # This tests the alternative branch using eigh().
+    n = 5
+    vals = np.arange(1, n+1, dtype=np.float64)
+    A = spdiags(vals, 0, n, n)
+    X = np.random.rand(n, 2)
+    lvals5, lvecs5 = lobpcg(A, X, largest=True, maxiter=100)
+    vals5, vecs5 = eigs(A, k=2)
+
+    _check_eigen(A, lvals5, lvecs5)
+    assert_allclose(vals5, lvals5, atol=1e-14)
+
+def test_verbosity():
+    """Check that nonzero verbosity level code runs.
```

Comment:
so we're not asserting anything here--nothing gets printed because the matrices are sufficiently Hermitian?

Reply:
The idea behind this test was to run the code paths that are not normally run in the default (silent) operation - there are other print statements in lobpcg(). So yes, we are not asserting anything. And actually, for the current settings of a=3, b=-1 in _report_nonhermitian() calls the gramA matrix is reported non-Hermitian. This could be fixed either by tweaking the parameters, or removing the _report_nonhermitian() completely. A cleaner solution how to provide optional diagnostics to users might be to support callbacks at suitable places in lobpcg() - what do you think?

Reply:
I'm not sure I now enough about the issue to ask for any substantive additional changes--I might politely suggest maybe just asserting that you get an expected result when probing the verbosity handling, but maybe not crucial.

Reply:
I have changed the verbosity level to 11 to make sure really all related code is executed. Subsequently, I have fixed two bugs. The code runs now without errors and this is the expected result - I am not sure how to assert that in the test? :)